hibernated_pindexes.go

```diff
@@ -0,0 +1,166 @@
+//  Copyright 2016-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package cbft
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/blevesearch/bleve/v2"
+	"github.com/blevesearch/bleve/v2/mapping"
+	index "github.com/blevesearch/bleve_index_api"
+	"github.com/couchbase/cbgt"
+	log "github.com/couchbase/clog"
+)
+
+var hibernatedPIndexUnimplementedErr = errors.New("unimplemented for " +
+	"hibernated pindexes")
+
+type HibernatedPIndex struct {
+	name   string
+	pindex *cbgt.PIndex
+	mgr    *cbgt.Manager
+}
+
+func (m *HibernatedPIndex) Name() string {
+	return m.name
+}
+
+func (m *HibernatedPIndex) SetName(name string) {
+	m.name = name
+}
+
+func (m *HibernatedPIndex) Index(id string, data interface{}) error {
+	return hibernatedPIndexUnimplementedErr
+}
+
+func (m *HibernatedPIndex) Delete(id string) error {
+	return hibernatedPIndexUnimplementedErr
+}
+
+func (m *HibernatedPIndex) Batch(b *bleve.Batch) error {
+	return hibernatedPIndexUnimplementedErr
+}
+
+func (m *HibernatedPIndex) Document(id string) (index.Document, error) {
+	return nil, hibernatedPIndexUnimplementedErr
+}
+
+func (m *HibernatedPIndex) DocCount() (uint64, error) {
+	return 0, nil
+}
+
+func (m *HibernatedPIndex) Search(req *bleve.SearchRequest) (
+	*bleve.SearchResult, error) {
+	return m.SearchInContext(context.Background(), req)
+}
+
+func (m *HibernatedPIndex) SearchInContext(ctx context.Context,
+	req *bleve.SearchRequest) (*bleve.SearchResult, error) {
+	log.Printf("hib pindexes: in searchincontext...")
+
+	restart := func() {} // just for trial
+	bleveParams := NewBleveParams()
+	kvConfig, _, _ := bleveRuntimeConfigMap(bleveParams)
+	kvConfig["read_only"] = true
+	log.Printf("hib pindexes: start open using: %s", m.pindex.Path)
+	bindex, err := bleve.OpenUsing(m.pindex.Path, kvConfig)
+	if err != nil {
+		return nil, fmt.Errorf("hib pindexes: err opening pindex path: %s", err.Error())
+	}
+	// need to initialise the bleve-related parameters to avoid a nil bindex
+	m.pindex.Impl = bindex
+	m.pindex.Dest = &cbgt.DestForwarder{
```

Comment:
remove all the unused code... is this neeeded?

Reply:
Sorry, I thought it was needed to avoid an earlier error. Fixed in the latest commit.